snapcraft_legacy/storeapi/v2/validation_sets.py

```diff
@@ -72,6 +80,8 @@ class Snap(models.CraftBaseModel):
     revision: int | None = None
     """Snap revision"""
 
+    components: dict[str, str | Component] | None = None
```

Comment:
One more suggestion to ensure the shorthand notation gets validated correctly:

```suggestion
    components: dict[str, Literal["required", "optional", "invalid"] | Component] | None = None
```

Reply:
Thanks! How would you feel about something like:
```
Presence = Literal["required", "optional", "invalid"]
```
Could be used in those thee places?